template/.github/workflows/build.yml.j2

```diff
@@ -0,0 +1,303 @@
+  # It then runs a `git diff` and fails the entire workflow, if any difference is encountered.
+  #
+  # Since CRD files are generated during the 'cargo build' process we need to run this once after
+  # removing the CRD files to ensure that the checked in versions match what the code expects.
+  #
+  # The reason for this step is, that developers are expected to check in up-to-date versions of charts
+  # and manifests, as we'd otherwise have to build these in CI and commit them back to the PR, which
+  # creates all kinds of problems.
+  # Therefor this failsafe simply aborts anything that has not had charts and manifests rebuilt before pushing.
+  check_charts:
+    name: Check if committed Helm & Kustomize Charts were up to date
+    needs:
+      - run_cargodeny
+      - run_clippy
+      - run_rustfmt
+      - run_rustdoc
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Helm
+        uses: azure/setup-helm@v2.0
+        with:
+          version: v3.6.2
+      - name: Set up cargo
+        uses: actions-rs/toolchain@v1.0.7
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - name: Set up rust-cache
+        uses: Swatinem/rust-cache@v1.3.0
+        with:
+          key: charts
+      - name: Regenerate charts
+        run: make regenerate-charts
+      - name: Check if committed charts were up to date
+        run: git diff --exit-code
+      - name: Git Diff showed uncommitted changes
+        if: ${{ failure() }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('Committed charts were not up to date, please regenerate and re-commit!')
+
+  test_charts:
+    name: Run Chart Tests
+    needs:
+      - check_charts
+      - run_tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: placeholder
+        run: echo Tests will go here
+
+  package_and_publish:
+    name: Package Charts, Build Docker Image and publish them
+    needs:
+      - test_charts
+      - select_repo
+    runs-on: ubuntu-latest
+    env:
+      NEXUS_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+      REPO: ${{ needs.select_repo.outputs.repository }}
+    if: needs.select_repo.outputs.repository != 'skip'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        if: ${{ github.event_name == 'pull_request' }}
+      - uses: actions-rs/toolchain@v1.0.7
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt
+          override: true
+      - name: Install requirements for version tool
+        if: ${{ github.event_name == 'pull_request' }}
+        run: pip install -r python/requirements.txt
+        # This step checks if the current run was triggered by a push to a pr (or a pr being created).
```

Comment:
Newlines between the steps would make it easier to see here especially with the comments but that's a nit

Reply:
done